api/src/main/java/org/apache/iceberg/actions/Action.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.actions;
+
+import java.util.Map;
+
+/**
+ * An action performed on a table.
+ *
+ * @param <ThisT> the child Java API class, returned by method chaining
+ * @param <R> the Java type of the result produced by this action
+ */
+public interface Action<ThisT, R> {
```

Comment:
This interface is still implemented by old actions that we will deprecate.

Reply:
We intend to remove this eventually?

Reply:
I am not sure. I think it is reasonable to have a common parent even for new actions. What are your thoughts?

Reply:
Yeah, I think it's reasonable to have a common parent. Your comment made me think you planned to remove it.